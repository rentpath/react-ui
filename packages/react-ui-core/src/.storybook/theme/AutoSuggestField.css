.AutoSuggestField {
  display: flex;
  margin: 75px;
}

.AutoSuggestField .Dropdown,
.AutoSuggestField .Field {
  margin: 0;
  width: 300px;
}

.AutoSuggestField .Button {
  height: 40px;
  position: relative;
  left: -70px;
  top: 5px;
}

.AutoSuggestField .Card {
  height: auto;
  width: auto;
  text-align: left;
}

.AutoSuggestField .List {
  padding: 0;
}

.AutoSuggestField .ListItem {
  padding: 5px 0;
}

.AutoSuggestField .Highlighter-Match {
  background: none;
}

.AutoSuggestField .Menu {
  padding-left: 5px;
}

.clear .Button {
  left: -40px;
}
