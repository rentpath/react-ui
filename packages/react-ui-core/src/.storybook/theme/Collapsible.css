.Collapsible_Items-visible {
  display: block;
}

.Collapsible_Items-hidden {
  display: none;
}

.Collapsible .Label {
  min-width: 200px;
}
