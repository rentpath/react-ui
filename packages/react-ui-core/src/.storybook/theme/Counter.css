.Counter {
  display: flex;
  white-space: nowrap;
  padding: 10px;
  align-items: center;
}

.Counter .Label {
  display: inline-block;
  white-space: nowrap;
  padding-right: 10px;
}

.Counter_Controls {
  display: inline-block;
  white-space: nowrap;
}

.Counter .Text {
  display: inline-block;
  text-align: center;
  padding: 0 20px;
}

.Counter_Button span {
  color: white;
  border-radius: 5px;
  padding: 5px 15px;
  display: inline-block;
  background: #0175D7;
  cursor: pointer;
}

.Counter_Button .Button {
  border-radius: 50%;
  width: 38px;
}

.Counter .Button {
  color: #ffffff;
  font-weight: bold;
}
