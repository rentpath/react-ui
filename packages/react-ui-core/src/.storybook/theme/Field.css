.Input,
.Select,
.Textarea {
  background-color: #ffffff;
  box-shadow: 0 0.5px 1px 0 rgba(0, 0, 0, 0.08);
  border: solid 1px #e9e9e9;
  border-radius: 4px;
  padding: 16px 10px;
  font-size: 13px;
}

.Field {
  margin: 16px 0;
  display: flex;
  max-width: 600px;
  color: #000;
  align-items: center;
}

.Field .Label {
  padding-right: 10px;
}
