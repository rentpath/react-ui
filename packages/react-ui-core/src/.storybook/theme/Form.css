.Form {
  min-width: 500px;
}

.Form .Field {
  color:#000;
}

.Form .Text {
  text-transform: uppercase;
  padding-bottom: 10px;
  margin-bottom: 10px;
}

.Form .Label {
  flex: 1;
}

.Form .Input,
.Form .Select,
.Form .Textarea {
  flex: 4;
}
