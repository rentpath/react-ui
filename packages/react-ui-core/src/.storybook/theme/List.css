.List {
  display: flex;
  padding: 20px;
}

.List-horizontal {
  flex-direction: row;
  justify-content: space-around;
}

.List-horizontal .ListItem {
  width: 100px;
}

.List-vertical {
  flex-direction: column;
}

.ListItem {
  cursor: pointer;
}

.ListItem-highlight {
  color: #AAA;
}
