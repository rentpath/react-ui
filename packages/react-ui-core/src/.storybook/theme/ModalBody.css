.ModalBody {
  min-width: 300px;
  padding: 20px;
  background: #FFF;
  flex-direction: column;
  border-radius: 5px;
  box-shadow: 1px 1px 5px #666;
  position: relative;
}
