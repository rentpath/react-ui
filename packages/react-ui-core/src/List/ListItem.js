import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import cn from 'classnames'
import autobind from 'autobind-decorator'
import themed from 'react-themed'

@themed(/^ListItem/, {
  pure: true,
})

export default class ListItem extends PureComponent {

  static propTypes = {
    theme: PropTypes.object,
    className: PropTypes.string,
    nodeType: PropTypes.oneOfType([
      PropTypes.node,
      PropTypes.func,
      PropTypes.object,
    ]),
    children: PropTypes.node,
    onMouseEnter: PropTypes.func,
    onClick: PropTypes.func,
    index: PropTypes.number,
    highlight: PropTypes.bool,
    selected: PropTypes.bool,
  }

  static defaultProps = {
    theme: {},
    nodeType: 'li',
  }

  @autobind
  handleMouseEnter() {
    const { onMouseEnter, index } = this.props

    if (onMouseEnter) onMouseEnter(index)
  }

  render() {
    const {
      theme,
      className,
      children,
      onMouseEnter,
      index,
      highlight,
      selected,
      nodeType: NodeType,
      ...props
    } = this.props

    return (
      <NodeType
        className={cn(
          theme.ListItem,
          className,
          highlight && theme['ListItem-highlight'],
          selected && theme['ListItem-selected'],
        )}
        index={index}
        onMouseEnter={this.handleMouseEnter}
        {...props}
      >
        {children}
      </NodeType>
    )
  }
}
