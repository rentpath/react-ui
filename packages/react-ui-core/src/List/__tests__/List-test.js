import React from 'react'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'
import theme from './mocks/theme'
import ThemedList from '../List'
import ListItem from '../ListItem'

const List = ThemedList.WrappedComponent
const items = [1, 2, 3]
const menuItems = [
  {
    label: 'one',
    value: 1,
    test_attr: '2',
  },
  {
    label: 'two',
    value: 2,
    test_attr: '2',
  },
]
const labellessItems = [
  {
    value: 1,
    test_attr: '2',
  },
  {
    value: 2,
    test_attr: '2',
  },
]

const functionItems = [
  () => (<div>foo</div>),
]

describe('List', () => {
  describe('mount', () => {
    const setup = props => shallow(<List theme={theme} {...props} />)

    it('applies a theme classname', () => {
      const className = setup().prop('className')
      expect(className).not.toEqual('List-')
    })

    it('passes through props to List Item', () => {
      const wrapper = mount(
        <List
          theme={theme}
          listItem={{ nodeType: 'span' }}
          items={items}
        />
      )
      expect(wrapper.find('ListItem').at(0).prop('nodeType')).toEqual('span')
    })

    it('passes through object props', () => {
      const snap = renderer
        .create(<List items={menuItems} />)
        .toJSON()
      expect(snap).toMatchSnapshot()
    })

    it('handles labelless items', () => {
      const errorFunc = global.console.error
      global.console.error = jest.fn()
      const snap = renderer
        .create(<List items={labellessItems} />)
        .toJSON()
      expect(snap).toMatchSnapshot()
      expect(console.error).toBeCalled() // eslint-disable-line no-console
      console.error = errorFunc // eslint-disable-line no-console
    })

    it('handles function items', () => {
      const snap = renderer
        .create(<List items={functionItems} />)
        .toJSON()
      expect(snap).toMatchSnapshot()
    })

    it('accepts custom node type', () => {
      const wrapper = setup({ nodeType: 'div' })
      expect(wrapper.find('div').length).toBe(1)
    })

    it('accepts custom orientation', () => {
      const wrapper = setup({ orientation: 'horizontal' })
      const className = wrapper.find('ul').prop('className')
      expect(className).toEqual('List-horizontal')
    })

    describe('highlightIndex', () => {
      it('sets ListItem `activate` true when highlightIndex equals index', () => {
        const highlightIndex = 1
        const wrapper = setup({ highlightIndex, items })
        wrapper.find(ListItem).map((item, index) => { // eslint-disable-line
          const active = index === highlightIndex
          expect(item.prop('highlight')).toEqual(active)
        })
      })

      it('does not set ListItem `activate` true when no prop passed', () => {
        const wrapper = setup({ items })
        wrapper.find(ListItem).map((item, index) => { // eslint-disable-line
          expect(item.prop('highlight')).toEqual(false)
        })
      })
    })
  })

  describe('prop change', () => {
    let wrapper

    beforeEach(() => {
      wrapper = shallow(<List theme={theme} />)
    })

    it('passes through props to List Item', () => {
      wrapper.setProps({ listItem: { nodeType: 'span' }, items })
      expect(wrapper.find('ul').childAt(0).prop('nodeType')).toEqual('span')
    })

    it('accepts custom node type', () => {
      wrapper.setProps({ nodeType: 'div' })
      expect(wrapper.find('div').length).toBe(1)
    })

    describe('orientation', () => {
      it('sets a default orienatation class', () => {
        const className = wrapper.find('ul').prop('className')
        expect(className).toEqual('List-vertical')
      })

      it('adds a className when passed', () => {
        wrapper.setProps({ orientation: 'horizontal' })
        const className = wrapper.find('ul').prop('className')
        expect(className).toEqual('List-horizontal')
      })
    })

    describe('highlightIndex', () => {
      it('sets ListItem `highlight` true when highlightIndex equals index', () => {
        const highlightIndex = 1
        wrapper.setProps({ highlightIndex, items })
        wrapper.find(ListItem).forEach((item, index) => {
          const active = index === highlightIndex
          expect(item.prop('highlight')).toEqual(active)
        })
      })
    })

    describe('selectedIndex', () => {
      it('sets ListItem `selected` true when highlightIndex equals index', () => {
        const selectedIndex = 1
        wrapper.setProps({ selectedIndex, items })
        wrapper.find(ListItem).forEach((item, index) => {
          const active = index === selectedIndex
          expect(item.prop('selected')).toEqual(active)
        })
      })
    })
  })
})
