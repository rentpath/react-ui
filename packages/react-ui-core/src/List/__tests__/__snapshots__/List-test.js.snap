// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List mount handles function items 1`] = `
<ul
  className=""
>
  <li
    className=""
    data-tid="list-item-0"
    index={0}
    onMouseEnter={[Function]}
  >
    <div>
      foo
    </div>
  </li>
</ul>
`;

exports[`List mount handles labelless items 1`] = `
<ul
  className=""
>
  <li
    className=""
    data-tid="list-item-0"
    index={0}
    onMouseEnter={[Function]}
    test_attr="2"
    value={1}
  />
  <li
    className=""
    data-tid="list-item-1"
    index={1}
    onMouseEnter={[Function]}
    test_attr="2"
    value={2}
  />
</ul>
`;

exports[`List mount passes through object props 1`] = `
<ul
  className=""
>
  <li
    className=""
    data-tid="list-item-0"
    index={0}
    onMouseEnter={[Function]}
    test_attr="2"
    value={1}
  >
    one
  </li>
  <li
    className=""
    data-tid="list-item-1"
    index={1}
    onMouseEnter={[Function]}
    test_attr="2"
    value={2}
  >
    two
  </li>
</ul>
`;
